### ISSUE TITLE
Sometimes unable to start SP world

### ISSUE THREAD

**Issue by Prunebane** [GTNewHorizons/GT-New-Horizons-Modpack#5984]
#### Which modpack version are you using?
2.0.8.8
(Current modification: twilightforest-1.7.10-2.3.8.3-dev.jar
though it happened even with unchanged 2.0.8.3 and 2.0.8.8)

#
#### If in multiplayer; On which server does this happen? 
Singleplayer

#
#### What did you try to do, and what did you expect to happen?
Started the game, started my SP world. Expected to load in.

#
#### What happened instead? (Attach screenshots if needed)
Game went through "Loading chunks" phase, then went back to Minecraft's main menu.
In that instance, you're unable to play singleplayer, until you turn Minecraft off, and back on.

Attaching all log files that I found. (this problem didn't trigger any crashlog!)
[logs.zip](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/4534730/logs.zip)

Also attaching what I was able to extract from the Technic console [console_output.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/4534728/console_output.txt)

it truncates after some size, so not complete - but, there is a "break point" I added here, look for the string:
<<Clicked to start playing the world, again>>
It's at that point that booting the world stopped, and you can see that trying to start it again actually turns off the internal server before doing anything else.

This issue is **difficult to reproduce**. I am unable to trigger it intentionally. It seems to happen when the modpack is starting in the background, but that is honestly just an assumption. My laptop is not the greatest, barely getting 60 FPS with view distance 8 chunks. (vanilla MC is fine, FPS-wise)

**Comment by Prunebane** [comment]
Additional excerpt of a log from a successful startup: 
[console_output_OK.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/4535172/console_output_OK.txt)

This hints at an issue with Netty, perhaps?

**Comment by Prometheus0000** [comment]
My understanding is that this is a problem with minecraft you just have to deal with, and is not specific to this pack, or even modded.

**Comment by Prunebane** [comment]
I see, thanks for the info.

Any chance you could link me a source towards this issue? Or hints what to look for? (if possible)

**Comment by Prometheus0000** [comment]
I wish I could, I have no idea where to find real information either :(. Try 'the internet', the minecraft bug report website or the discord I guess? Or maybe someone who knows more can chime in?

**Comment by quedral** [comment]
try to adjust ram settings like minimum:2gigs maximum 6-8 gigs. 

giving too much or too less ram can cause this issue.

**Comment by Prunebane** [comment]
I'm currently with 6 gigs of RAM, I may try and go down to 5 - that was the bare minimum I needed to run somewhat smoothly, see if this does anything.
I might pop up back in if I have anything new with this change

**Comment by Dream-Master** [comment]
4 is minimum

**Comment by GTNH-Afx237v7** [comment]
I‘ve had the same issue few times as well. However, sometimes it would take a moment (few minutes) to finally load the world.
Are you using any java arguments, if so, which? My computer is also not the best, however adjusting the JVM arguments helped me a lot.

**Comment by Prunebane** [comment]
I haven't changed JVM arguments - using Technic launcher, it is possible that there are some pre-defined arguments though.
Do share yours, I can try and see if anything changes. (may take over a week, though - I will be busy the next couple days, less time for GTNH)

**Comment by GTNH-Afx237v7** [comment]
Here are the JVM arguments that I use:
-d64 -XX:+UseG1GC -Xmx4G -Xms4G -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -XX:ParallelGCThreads=4

**Comment by Dream-Master** [comment]
@Prunebane any update here ?

**Comment by Prunebane** [comment]
Just found the arguments, currently booted fine (once) without them - only change was from 6GB to 5GB RAM. Technic launcher, just to be complete.
As I said, it doesn't happen always, it's like a 1 in 10 chance after bootup.
I'll definitely return here when I get a new crash without arguments - will add them once that happens, see if that changes anything.

ETA for more info - probably by the end of this week (if it happens again)

**Comment by Prunebane** [comment]
Happened again. Adding recent logs.
[28_04_20.zip](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/4543379/28_04_20.zip)

Two things out of the ordinary(?) that I've noticed.
- There was a set of spam this time around in the earliest stage of booting the modpack (phase 1 I think) about each mod's "version" unknown, defaulting to some value. I don't see this in regular logs, did see it in console stdout.
((Trying to compare with an OK log, found that MCVersion isn't found in some cases; however, both the bad and the good logs have 25 hits on this line, so this is probably OK, just me never noticing it before..?))
- Trying to look through the latest (bad) log and the previous (ok) log, looking for the above thing, I noticed one difference - GalaxySpace is complaining about some version of a library:
`[06:05:15] [Client thread/INFO] [STDOUT/GalaxySpace]: [galaxyspace.SolarSystem.core.animations.MCACommonLibrary.MCAVersionChecker:checkForLibraryVersion:11]: MCA WARNING: galaxyspace.BarnardsSystem.planets.barnardaC.models.ModelTentacles needs a newer version of the library (5). Things could go wrong!
[06:05:15] [Client thread/INFO] [STDOUT/GalaxySpace]: [galaxyspace.SolarSystem.core.animations.MCACommonLibrary.MCAVersionChecker:checkForLibraryVersion:11]: MCA WARNING: galaxyspace.BarnardsSystem.planets.barnardaC.models.ModelCrawler needs a newer version of the library (5). Things could go wrong!
`

Added all relevant recent logs, and added one previous FML log (from an OK boot), just for comparison.
